applications/StructuralMechanicsApplication/custom_elements/membrane_elment.cpp

```diff
@@ -0,0 +1,1100 @@
+    rTransformationMatrix(0,2) = 2.0*e_g_11*e_g_12;
+    rTransformationMatrix(1,0) = e_g_21*e_g_21;
+    rTransformationMatrix(1,1) = e_g_22*e_g_22;
+    rTransformationMatrix(1,2) = 2.0*e_g_21*e_g_22;
+    rTransformationMatrix(2,0) = e_g_11*e_g_21;
+    rTransformationMatrix(2,1) = e_g_12*e_g_22;
+    rTransformationMatrix(2,2) = (e_g_11*e_g_22) + (e_g_12*e_g_21);
+}
+
+void MembraneElement::TransformStrains(Vector& rStrains,
+  Vector& rReferenceStrains, const Matrix& rTransformationMatrix)
+{
+    // use contravariant basevectors here
+    // transform base vecs needs only G3 which is equal for co and contra if it is orthogonal
+    // tranform strains needs contra-variant !
+    rStrains = ZeroVector(3);
+    rReferenceStrains[2]/=2.0; // extract E12 from voigt strain vector
+    rStrains = prod(rTransformationMatrix,rReferenceStrains);
+    rStrains[2]*=2.0; // include E12 and E21 for voigt strain vector
+}
+
+void MembraneElement::AddPreStressPk2(Vector& rStress, const array_1d<Vector,2>& rTransformedBaseVectors){
+
+    Vector pre_stress = ZeroVector(3);
+    if (GetProperties().Has(PRESTRESS_VECTOR)){
+        pre_stress = GetProperties()(PRESTRESS_VECTOR);
+
+        if (Has(LOCAL_PRESTRESS_AXIS_1) && Has(LOCAL_PRESTRESS_AXIS_2)){
+
+            array_1d<array_1d<double,3>,2> local_prestress_axis;
+            local_prestress_axis[0] = GetValue(LOCAL_PRESTRESS_AXIS_1)/MathUtils<double>::Norm(GetValue(LOCAL_PRESTRESS_AXIS_1));
+            local_prestress_axis[1] = GetValue(LOCAL_PRESTRESS_AXIS_2)/MathUtils<double>::Norm(GetValue(LOCAL_PRESTRESS_AXIS_2));
+
+            Matrix transformation_matrix = ZeroMatrix(3);
+            InPlaneTransformationMatrix(transformation_matrix,rTransformedBaseVectors,local_prestress_axis);
+            pre_stress = prod(transformation_matrix,pre_stress);
+
+        } else if (Has(LOCAL_PRESTRESS_AXIS_1)) {
+
+            Vector base_3 = ZeroVector(3);
+            MathUtils<double>::CrossProduct(base_3, rTransformedBaseVectors[0], rTransformedBaseVectors[1]);
+            base_3 /= MathUtils<double>::Norm(base_3);
+
+            array_1d<array_1d<double,3>,2> local_prestress_axis;
+            local_prestress_axis[0] = GetValue(LOCAL_PRESTRESS_AXIS_1)/MathUtils<double>::Norm(GetValue(LOCAL_PRESTRESS_AXIS_1));
+
+            MathUtils<double>::CrossProduct(local_prestress_axis[1], base_3, local_prestress_axis[0]);
+            local_prestress_axis[1] /= MathUtils<double>::Norm(local_prestress_axis[1]);
+
+            Matrix transformation_matrix = ZeroMatrix(3);
+            InPlaneTransformationMatrix(transformation_matrix,rTransformedBaseVectors,local_prestress_axis);
+            pre_stress = prod(transformation_matrix,pre_stress);
+        }
+    }
+    rStress += pre_stress;
+}
+
+void MembraneElement::StressPk2(Vector& rStress,
+    const Matrix& rReferenceContraVariantMetric,const Matrix& rReferenceCoVariantMetric,const Matrix& rCurrentCoVariantMetric,
```

Comment:
I think rReferenceContraVariantMetric is not used in this function so I guess we don't need to pass it.